### PR TITLE
a8n: Show read-only campaign fields in a grid view

### PR DESCRIPTION
**Pull request by eseliger**
To make more clear what is read only and what will be editable when using the edit button.

### CODE REVIEW

**Comment by tsenart** [comment]
In PRs like this it'd be cool to have a before / after screenshot in the PR description. What do you think?

**Comment by eseliger** [comment]
You can find it in Percy :)

**Comment by tsenart** [comment]
I only see red squiggles in the new version. What am I doing wrong?

**Comment by eseliger** [comment]
Clicking on the red diff reveals the whole picture

File: web/src/enterprise/campaigns/detail/CampaignDetails.tsx [hunk 1]
```diff
@@ -72,6 +72,12 @@ const defaultInputByType: { [K in CampaignType]: any } = {
 }`,
 }
 
+const typeLabels: { [K in CampaignType | '']: string } = {
```
Comment (lguychard) [inline web/src/enterprise/campaigns/detail/CampaignDetails.tsx]:
```suggestion
const typeLabels: Record<CampaignType | '', string> = {
```

File: web/src/enterprise/campaigns/detail/CampaignDetails.tsx [hunk 2]
```diff
@@ -87,7 +93,7 @@ export const CampaignDetails: React.FunctionComponent<Props> = ({
     const [name, setName] = useState<string>('')
     const [description, setDescription] = useState<string>('')
     const [type, setType] = useState<CampaignType>()
-    const [campaignPlanArguments, setCampaignPlanArguments] = useState<string>('')
+    const [campaignPlanArguments, setCampaignPlanArguments] = useState<string | undefined>()
```
Comment (lguychard) [inline web/src/enterprise/campaigns/detail/CampaignDetails.tsx]: The `| undefined` is implicit and correctly inferred when an argument for the initial value is not passed.

```suggestion
    const [campaignPlanArguments, setCampaignPlanArguments] = useState<string>()
```

**Pull request merged**